SC_redesign/Crop_and_Soil/crop/phosphorus_uptake.py

```diff
@@ -0,0 +1,316 @@
+from math import log, exp
+from typing import List
+from SC_redesign.Crop_and_Soil.soil.soil import Soil
+
+class PhosphorusUptake:
+    def __init__(self):
+
+        #these attrs DO need initial values
+        self.actual_phosphorus_biomass = None       #bio_P
+        self.potential_phosphorus_uptake = None     #P_up
+        self.mature_fractional_phosphorus = None    #fr_p3
+        self.potential_phosphorus_biomass_increase = None  #d_biomass_max    ##i think clay has a diff name for this
+        self.previous_actual_biomass = None         #prev_biomass_actual
+        self.emergent_fractional_phosphorus = None  #fr_p1
+        self.previous_fraction_potential_heat_units = None #prev_fr_PHU
+        self.fraction_potential_heat_units_half_maturity = None #fr_PHU_50
+        self.half_mature_fractional_phosphorus = None #fr_p2
+        self.fraction_potential_heat_units_full_maturity = None #fr_PHU_100
+        self.near_mature_fractional_phosphorus = None #fr_p3ish
+
+
+        #these attrs are to be set later, start out as None
+        self.fractional_biomass_phosphorus = None   #fr_P
+        self.optimal_phosphorus_biomass = None      #bio_P_opt
+
+
+    def phosphorus_uptake_driver(self,soil: Soil):
```

Comment:
give your functions more descriptive names that start with verbs. Since this is the main function for this class, try naming it similar to the class. 
```suggestion
    def incorporate_phosphorus(self,soil: Soil):
```